### ISSUE TITLE
loginByCookie: Need to switch database BEFORE this method fires up

### ISSUE THREAD

**Issue by Commifreak** [yiisoft/yii2#14812]
### What steps will reproduce the problem?
I have multiple database connections and therefore multiple user tables. I use one "MasterUser" model which is declared in my config as:

```php
'components' => [
            'user' => [
            'class' => 'yii\web\User',
            'identityClass' => 'common\models\MasterUser',
```
The getDB-function in this model determines, which db-selector to use.

Now we have autologin enables as well:

```php
'enableAutoLogin' => true,
```
The cookies is here, the authKey is ok, BUT: After close the browser and revisiting the site, my old session is gone. I have a session-based variable which helps me to determine the database connections on base of a login-dropdown field while manually login.

It would be nice to make some stuff BEFORE User::loginByCookie fires up. The handleBeforeLogin Event is triggered AFTER loginByCookie.

### What is the expected result?
Some way I could run a code one time before loginByCookie fires up.

### What do you get instead?
loginByCookie does not use the correct database and resets the cookie.

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.12
| PHP version      | 5.5.9
| Operating system | Ubuntu 14.04

**Comment by Commifreak** [comment]
Ok, if someone is interested: I made a workaround:

```php
'session' => [
             ...
            'class' => 'common\components\Session'
             ...
        ]
```

```php    class Session extends \yii\web\Session {
    /**
     * This method overrides Session::open to set some things before the User::loginbyCookie Method fires up.
     */
    public function open() {
        parent::open();
        $value = Yii::$app->getRequest()->getCookies()->getValue(Yii::$app->user->identityCookie['name']);
        if(!isset($_SESSION['loginByCookieWorkaround']) && $value !== null && !isset($_SESSION['client'])) {
            Yii::info('Workaround: User is logged out but has valid cookie. Client assignment needed.');

            $data = json_decode($value, true);
            list($id, $authKey, $duration) = $data;

            foreach(Yii::$app->components as $component => $compValue) {
                if(strpos($component, 'db_client') === 0) {
                    Yii::trace('Found a match! => '.$component);

                    $client = substr($component, strlen('db_client_'));
                    Yii::trace('Will work with client: '.$client);
                    $_SESSION['client'] = $client;

                    // Now that the client-SESSION variable is set, the Model::getDB function selects another db
                    $user = User::findOne($id);
                    if($user && $user->auth_key == $authKey) {
                        Yii::trace('Yep, got right client - well, at least the userid and authkey macthes, so the possibility is very large that we got the right user.');
                        return true;
                    } else {
                        Yii::trace('Not the right client :(');
                        unset($_SESSION['client']);
                        continue;
                    }
                }
            }


            $_SESSION['loginByCookieWorkaround'] = true;
        }

    }
}
```

With that code, I create the base of database-selecton logic: session['client']. This makes loginByCookie working again ;)

However, a native possibility to do things once before loginByCookie() fires up would be nice to have.

**Comment by samdark** [comment]
Won't be implemented in 2.0.
